src/DynamoCoreWpf/ViewModels/Watch3D/Watch3DViewModelBase.cs

```diff
@@ -449,6 +453,36 @@ protected virtual void OnRenderPackagesUpdated(NodeModel node, IEnumerable<IRend
             // Override in derived classes
         }
 
+        internal event Func<MouseEventArgs, Ray3D> RequestClickRay;
+        public Ray3D GetClickRay(MouseEventArgs args)
+        {
+            if (RequestClickRay != null)
+                return RequestClickRay(args);
+
+            return null;
+        }
+
+        public event Action<object, MouseButtonEventArgs> ViewMouseDown;
+        internal void OnViewMouseDown(object sender, MouseButtonEventArgs e)
+        {
+            if (ViewMouseDown != null)
+                ViewMouseDown(sender, e);
+        }
```

Comment:
Minor thing, but it is usually recommended to do this instead for all event handler invokes:

```csharp
var handler = ViewMouseDown;
if (handler) handler(sender, e);
```

That's to prevent `ViewMouseDown` being reset by other threads before it is invoked.